hamilton/function_modifiers/recursive.py

```diff
@@ -423,7 +423,29 @@ def _derive_name(self, fn: Callable) -> str:
 
     def generate_nodes(self, fn: Callable, configuration: Dict[str, Any]) -> Collection[node.Node]:
         # Resolve all nodes from passed in functions
-        resolved_config = dict(configuration, **self.config)
+        # if self.config has source() or value() in it, we need to resolve it
+        sources_to_map = {}
```

Comment:
This should be it's own function `resolve_config` -- then we can test it directly

Reply:
sure. done